lib/Driver.js

```diff
@@ -90,44 +90,60 @@ module.exports = class Driver {
   /**
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
-   * @param {Ed25519KeyPair} edKey
+   * @param {Ed25519KeyPair} ed25519Key
    * @returns {DidDocument}
    */
-  async keyToDidDoc(edKey) {
+  async keyToDidDoc(ed25519Key) {
+    const edKey = new Ed25519KeyPair({
+      publicKeyBase58: ed25519Key.publicKey,
+      privateKeyBase58: ed25519Key.privateKey,
+    });
     const did = `did:key:${edKey.fingerprint()}`;
-    const keyId = `${did}#${edKey.fingerprint()}`;
+    edKey.id = `${did}#${edKey.fingerprint()}`;
     edKey.controller = did;
 
     const dhKey = await X25519KeyPair.fromEdKeyPair(edKey);
     dhKey.id = `${did}#${dhKey.fingerprint()}`;
+    dhKey.controller = did;
+
+    // get the public components of each keypair
+    const publicEdKey = this.getPublicVerificationMethod({key: edKey});
+    const publicDhKey = this.getPublicVerificationMethod({key: dhKey});
 
+    // generate the DID Document
     const didDoc = {
       '@context': ['https://w3id.org/did/v0.11'],
       id: did,
-      publicKey: [{
-        id: keyId,
-        type: edKey.type,
-        controller: did,
-        publicKeyBase58: edKey.publicKeyBase58
-      }],
-      authentication: [keyId],
-      assertionMethod: [keyId],
-      capabilityDelegation: [keyId],
-      capabilityInvocation: [keyId],
-      keyAgreement: [{
-        id: dhKey.id,
-        type: dhKey.type,
-        controller: did,
-        publicKeyBase58: dhKey.publicKeyBase58
-      }]
+      publicKey: [publicEdKey], // FIXME: publicKey is deprecated, remove
+      verificationMethod: [publicEdKey],
+      authentication: [publicEdKey.id],
+      assertionMethod: [publicEdKey.id],
+      capabilityDelegation: [publicEdKey.id],
+      capabilityInvocation: [publicEdKey.id],
+      keyAgreement: [publicDhKey],
     };
+    // FIXME: 'keys' is misleading as devs might think 'public keys'; remove
     Object.defineProperty(didDoc, 'keys', {
       value: {
-        [keyId]: edKey,
+        [edKey.id]: edKey,
         [dhKey.id]: dhKey
       },
       enumerable: false
     });
+    Object.defineProperty(didDoc, 'keypairs', {
+      value: {
+        [edKey.id]: edKey,
+        [dhKey.id]: dhKey,
```

Comment:
This is known to be an anti-pattern for the hidden class optimizer. If we want to support getting keys by ID (which is a mostly random information) we should use a Map.

I'm also not sure how I feel about creating special non-enumerable properties (direct accessors) on this class that don't actually appear in the actual DID Document; it may confuse people. I think a method would be better.